src/zodios-error.utils.ts

```diff
@@ -0,0 +1,61 @@
+import { AxiosError } from "axios";
+import { findEndpointErrorByAlias, findEndpointErrorByPath } from "./utils";
+import { Merge } from "./utils.types";
+import { ZodiosError, ZodiosMatchingErrorType } from "./zodios-error";
+import {
+  Method,
+  ZodiosAliases,
+  ZodiosEndpointDefinitions,
+  ZodiosEndpointError,
+  ZodiosMatchingErrorsByAlias,
+  ZodiosMatchingErrorsByPath,
+  ZodiosPathsByMethod,
+} from "./zodios.types";
+
+function isDefinedError(
```

Comment:
I think it's proper to parse the error while [resolving response](https://github.com/ecyrbe/zodios/blob/ec792840f5ce99bf70b57fb6da57bd5fa3e52cb2/src/zodios.ts#L279).

It could be like:
```js
try {
  const res = await response;
} catch (err) {
  // if thrown, it's the errors we not bother to handle at this level (like memory error)
  if (!axios.isAxiosError(err)) throw err;
  if (!err.response) throw err;

  // more on getEndpointErrors later
  const errorSchemas = getEndpointErrors(api, method, path, err.response.status);
  if (errorSchemas.some((schema) => schema.safeParse(err.response.data).success)) {
    throw new ExpectedError(err);
  } else {
    throw new UnknownError(err); // or just throw err
  }
}
```

(To simplify it, I'm not considering alias)
Then `isDefinedError` only checks if `error instanceof ExpectedError` and is produced by given `method` and `path`. But NOT parsing the `error`

```ts
function isDefinedError(err: unknown, method: string, path: string): boolean {
  if (!(err instanceof ExpectedError)) return false;
  return !!getEndpointErrors(api, err.config.method, err.config.url, err.response.status).length;
}
```

### Why `getEndpointErrors` (plural) instead of [`findEndpointError`](https://github.com/ecyrbe/zodios/blob/ec792840f5ce99bf70b57fb6da57bd5fa3e52cb2/src/utils.ts#L84) (single)

```ts
function getEndpointErrors(
  api: ZodiosEndpointDefinitions,
  method: string,
  path: string,
  err: AxiosError
): ZodiosEndpointError[];
```

1. So we can define multiple error with the same status code (and I really need this)
```ts
makeErrors([
  { status: 400, schema: z.object({ error: z.enum(["invalid_format"]) }) }
  {
    status: 400,
    schema: z.object({
      error: z.enum(["fail_to_lock_the_resource", "require_lock_timeout"])
    })
  }
])
```
2. _Determining which schema the thrown error is_ is a duty of zodios user, not zodios.